### ISSUE TITLE
[Bug] AidenLx's Folder Note plugin (or dependencies) can cause Front Matter Title to fail to update File Explorer on startup

### ISSUE THREAD

**Issue by gbiagini** [Snezhig/obsidian-front-matter-title#42]
It looks like, when starting Obsidian, if AidenLx's Folder Note, Folder Note Core, and the AidenLx's Folder Note - folderv Component plugins are installed, Front Matter Title will not actually change the display of the File Explorer. It could be just a single one, I didn't test them individually.

The titles in the explorer will update when you go to the settings for the Front Matter Title plugin, but that seems to be the only way to trigger the update. Uninstalling the conflicting plugins solves the problem.

My guess is that the Front Matter Title plugin loads first (it had a faster startup time when I checked the Developer console, IIRC), then the update is overridden by AidenLx Folder Note, causing the original titles to appear. The issue seems to be limited to the File Explorer.

I'll try to dig into it more if I get some extra time. This could probably be fixed very easily by adding a 10-100ms sleep toggle option in the settings that can be enabled if users experience problems.

**Comment by Snezhig** [comment]
Hello.
You have many plugins, so it is difficult to say something.
But, I guess addition a command to restart plugin or re-update titles.
Also option to set delay before start all logic is a good idea too.

**Comment by Snezhig** [comment]
@gbiagini, Hi, you can try [ beta version 2.0.0](https://github.com/Snezhig/obsidian-front-matter-title/releases/tag/2.0.0-beta), using [BRAT](https://github.com/TfTHacker/obsidian42-brat). 
It brings delay option.

**Comment by gbiagini** [comment]
tl;dr - A delay of 1000ms (1 second) was sufficient to solve this issue and #40.

I stopped using the plugins - I'm new to Obsidian, so I've mostly been exploring. I really appreciate your help in addressing this though, so I temporarily reinstalled them and installed the beta to test it.

A delay of 1000ms (1 second) was sufficient to solve this issue. Interestingly, I had a suspicion the cause is the same for this and #40, given that the reporter describes a refresh as solving the issue. The 1000ms=1s delay is sufficient to resolve that issue as well - the Supercharged Links color scheme will persist if the startup delay is long enough.

I suspect that the necessary delay could vary based on installed plugins and user hardware, maybe also size/number of files in a vault.

**Comment by Cammagno** [comment]
I confirm the problem (both with Folder Note and Supercharged Links) and also that the solution is working with a delay of 1000.
Maybe it could be a good idea to set a delay greater than 0 as default value, considering how much these plugins are used (and that other ones may be involved).

**Comment by Snezhig** [comment]
@Cammagno, yes. That seems nice.

**Comment by Snezhig** [comment]
It seems solved, so I close the issue. Feel free to reopen if it is needed
